cause_go113.go

```diff
@@ -0,0 +1,37 @@
+// +build go1.13
+
+package errors
+
+import "errors"
+
+// Cause returns the underlying cause of the error, if possible.
+// An error value has a cause if it implements the following
+// interface:
+//
+//     type causer interface {
+//            Cause() error
+//     }
+//
+// If the error does not implement the Cause interface, the Cause
+// function will fallback to the standard library's errors.Unwrap
```

Comment:
Let’s be more explicitly clear about the contract we are building: will we return the original error of any arbitrary ordering of `Cause` and `Unwrap`? The text presented here is potentially ambiguous… hm, I think the original was somewhat ambiguous as well.

We should be clear that this function will keep working recursively until it finds an error that is both not a `causer` and would return `errors.Unwrap(err) == nil`.

Reply:
If an `error` has both `Cause()` and `Unwrap()`, I think `Cause()` should take precedence since that matches the original behavior of this function.

Reply:
That’s not what I am saying. If we have a `causer` wrapped by a `fmt.Errorf` wrapped error, which wraps a `causer`, what is the behavior?

The code as you have written would unwrap all layers (which I think it should), but we should document that it will recursively try unwrapping all layers until it gets an unwrapped error.